x-pack/plugins/maps/public/connected_components/edit_layer_panel/layer_settings/layer_settings.tsx

```diff
@@ -155,6 +162,70 @@ export function LayerSettings(props: Props) {
     );
   };
 
+  const renderShowLocaleSelector = () => {
+    if (!props.layer.supportsLabelLocales()) {
+      return null;
+    }
+
+    const options = [
+      {
+        text: i18n.translate('xpack.maps.layerPanel.settingsPanel.labelLanguageDefaultDropDown', {
+          defaultMessage: 'Default',
+        }),
+        value: DEFAULT_EMS_LOCALE,
+      },
+      {
+        text: i18n.translate(
+          'xpack.maps.layerPanel.settingsPanel.labelLanguageAutoselectDropDown',
+          {
+            defaultMessage: 'Autoselect based on Kibana locale',
+          }
+        ),
+        value: AUTOSELECT_EMS_LOCALE,
+      },
+      {
+        text: i18n.translate('xpack.maps.layerPanel.settingsPanel.labelLanguageEnglishDropDown', {
+          defaultMessage: 'English',
+        }),
+        value: 'en',
+      },
+      {
+        text: i18n.translate('xpack.maps.layerPanel.settingsPanel.labelLanguageChineseDropDown', {
+          defaultMessage: 'Chinese',
+        }),
+        value: 'zh-cn',
+      },
+      {
+        text: i18n.translate('xpack.maps.layerPanel.settingsPanel.labelLanguageJapaneseDropDown', {
+          defaultMessage: 'Japanese',
+        }),
+        value: 'ja-jp',
```

Comment:
Thanks @Bamieh. 

> I am not fully sure what you're trying to achieve in this feature but I want to note that we only load the kibana locale set by the i18n.locale config and the rest of the locales are not loaded.

Our Elastic Maps Service provides basemap reference data for Elastic Maps which includes place names in [many different languages](https://openmaptiles.org/languages/native/#1.32/0/0). So, unlike Kibana, all of the map label locales are always available. Currently we only show the default names which are usually in English. We hoped to expose more languages to our users and primarily focus on languages already supported in the Kibana UI. 

I think it will be best if we maintain a separate list of compatible locales for Maps rather than trying to directly support the discovered locales. If a custom Kibana locale does not match any of supported Maps locales, then the Maps will automatically fall back to the default language (English).